qmt/data/solvers_2d.py

```diff
@@ -2,20 +2,23 @@
 # Licensed under the MIT License.
 
 from collections import namedtuple
+from typing import NamedTuple, Tuple
+from sympy.core.mul import Mul
+import numpy as np
 
 Potential2dData = namedtuple('Potential2dData', ['coordinates', 'coordinate_units', 'potential',
                                                  'potential_units'])
-ThomasFermi2dData = namedtuple('ThomasFermi2dData', ['coordinates',
-                                                     'potential',
-                                                     'density',
-                                                     'conduction_band',
-                                                     'valence_band',
-                                                     'reference_level',
-                                                     'temperature',
-                                                     'coordinate_units',
-                                                     'potential_units',
-                                                     'energy_units'
-                                                     ])
+
+
+class ThomasFermi2dData(NamedTuple):
+    coordinates: Tuple[np.ndarray, np.ndarray]
+    potential: np.ndarray
+    density: np.ndarray
+    conduction_band: Mul
+    valence_band: Mul
+    reference_level: Mul
+    temperature: Mul
```

Comment:
why Mul here?

Reply:
That's the type for `1 * u.m`. Sympy takes that as a multiplication

Reply:
Unfortunately I don't think we can get more specific with the types here (python isn't really built for that :/). But within the task I do check that the units are correct (i.e. `to_float(input/u.the_unit_it's_supposed_to_be)`) doesn't throw. I also checked that expressions such as `10 * spu.m + 1 * spu.km` and `2/ spu.m ** -2` all have type Mul

Reply:
conduction band will be an array though. I remember you want to have every array with it's type, but it seems then that specifying types doesn't do much - it is Mul anyway.

Reply:
Well at least it tells you that it's supposed to have a unit attached?

Oh right the arrays are of type np.ndarray. Fixed them now